### ISSUE TITLE
occasional null ref in ServerSentEventsTransport.<OnStart>

### ISSUE THREAD

**Issue by BrannonKing** [SignalR/SignalR#3376]
I'm occasionally seeing the exception shown below in my application logs. I cannot tell that it has a negative impact nor what is causing it. I have no idea how to reproduce it at will.

```
System.NullReferenceException: Object reference not set to an instance of an object.
   at Microsoft.AspNet.SignalR.Client.Transports.ServerSentEventsTransport.<OnStart>b__0()
   at System.Action.Invoke()
   at Microsoft.AspNet.SignalR.Client.Infrastructure.TransportInitializationHandler.<>c__DisplayClassa.<CompleteFail>b__9()
   at Microsoft.AspNet.SignalR.Client.Infrastructure.TransportInitializationHandler.<>c__DisplayClassd.<Dispatch>b__c(Object _)
   at System.Threading.QueueUserWorkItemCallback.WaitCallback_Context(Object state)
   at System.Threading.ExecutionContext.RunInternal(ExecutionContext executionContext, ContextCallback callback, Object state, Boolean preserveSyncCtx)
   at System.Threading.ExecutionContext.Run(ExecutionContext executionContext, ContextCallback callback, Object state, Boolean preserveSyncCtx)
   at System.Threading.QueueUserWorkItemCallback.System.Threading.IThreadPoolWorkItem.ExecuteWorkItem()
   at System.Threading.ThreadPoolWorkQueue.Dispatch()
   at System.Threading._ThreadPoolWaitCallback.PerformWaitCallback()
```

**Comment by moozzyk** [comment]
What version are you using? Can you post logs?

**Comment by BrannonKing** [comment]
I'm using version 2.1.2. I'll see if I can work up a better log file. I assume that you only need client-side logs?

**Comment by moozzyk** [comment]
Yes, only client logs.

**Comment by BrannonKing** [comment]
Here's a log file:
http://kingtown.net/Uploads/Mobius_2015-01-02--13-03-30.log

**Comment by BrannonKing** [comment]
I added a null check to my branch:
https://github.com/BrannonKing/SignalR/commit/646e7d6a8b65dae33385b2be3cec527eba63bc0a

I'm guessing a little bit at the location.

**Comment by moozzyk** [comment]
The client code that failed with the NRE has been changed in 2.2.0. @halter73 claims I actually fixed this ;) You can try a 2.2.0 client nightly build from here: https://www.myget.org/gallery/aspnetwebstacknightly or wait until 2.2.0 is shipped on NuGet. If the issue persists in 2.2.0 I will take a look at this again.

**Comment by BrannonKing** [comment]
Which branch has the 2.2 code?

**Comment by moozzyk** [comment]
You can use either dev or release - they should be in sync at this point

**Comment by halter73** [comment]
There have been some changes to the ServerSentEventsTransport since 2.1.2, but I'm pretty sure this bug is still there in [ServerSentEventsTransport.OnStartFailed](https://github.com/SignalR/SignalR/blob/c213737ffd41b616ae3071d22b35e06ba4cf0055/src/Microsoft.AspNet.SignalR.Client/Transports/ServerSentEventsTransport.cs#L56) which can be called before `_request` is set if the [`disconnectToken` is tripped](https://github.com/SignalR/SignalR/blob/c213737ffd41b616ae3071d22b35e06ba4cf0055/src/Microsoft.AspNet.SignalR.Client/Infrastructure/TransportInitializationHandler.cs#L59) before an SSE request is made.

**Comment by BrannonKing** [comment]
Ah, I see it too. Thanks for the updates. For lack of a better place to ask, what is the schedule for 2.2 and what can I do to further encourage the present list of pull requests getting into the 2.2?

**Comment by halter73** [comment]
Unfortunately, I don't expect any currently open pull requests to make it in for the 2.2.0 release we are planning for next week. I actually wanted my open pull request to make it in too.

Hopefully we can merge the PRs in for 2.2.1.

**Comment by moozzyk** [comment]
Fixed in 43d72ba
